pkg/block/indexheader/header.go

```diff
@@ -1,19 +1,35 @@
 package indexheader
 
 import (
+	"io"
+
+	"github.com/pkg/errors"
 	"github.com/prometheus/prometheus/tsdb/index"
 )
 
-// NotFoundRange is a range returned by PostingsOffset when there is no posting for given name and value pairs.
-// Has to be default value of index.Range.
-var NotFoundRange = index.Range{}
+// NotFoundRangeErr is an error returned by PostingsOffset when there is no posting for given name and value pairs.
+var NotFoundRangeErr = errors.New("range not found")
 
-// Reader is an interface allowing to read essential, minimal number of index entries from the small portion of index file called header.
+// Reader is an interface allowing to read essential, minimal number of index fields from the small portion of index file called header.
 type Reader interface {
+	io.Closer
+
+	// IndexVersion returns version of index.
 	IndexVersion() int
+
+	// PostingsOffset returns start and end offsets of postings for given name and value.
+	// end offset might be bigger than actual posting ending, but not larger then the whole index file.
+	// NotFoundRangeErr is returned when no index can be found for given name and value.
 	// TODO(bwplotka): Move to PostingsOffsets(name string, value ...string) []index.Range and benchmark.
-	PostingsOffset(name string, value string) index.Range
+	PostingsOffset(name string, value string) (index.Range, error)
+
+	// LookupSymbol returns string based on given reference.
+	// Error is return if the symbol can't be found.
 	LookupSymbol(o uint32) (string, error)
-	LabelValues(name string) []string
+
+	// LabelValues returns all label values for given label name or error if not found.
```

Comment:
The comment here seems a bit wrong: we return an error when an actual error had occurred, not when we haven't found any label values.